plugins/org.obeonetwork.dsl.uml2.design/src/org/obeonetwork/dsl/uml2/design/internal/services/ReconnectSwitch.java

```diff
@@ -84,14 +85,31 @@
 	private Element newPointedElement;
 
 	/**
+	 * Edge to reconnect.
+	 */
+	private DEdge edgeToReconnect;
+
+	/**
 	 * {@inheritDoc}
 	 */
 	@Override
 	public Element caseAssociation(Association association) {
 		if (isReconnectable(association)) {
 			if (RECONNECT_SOURCE == reconnectKind) {
-				final Property source = AssociationServices.INSTANCE.getSource(association);
-				source.setType((Type)newPointedElement);
+				if (association.getMemberEnds().size() > 2) {
+					final String sourceObject = edgeToReconnect.getSourceNode().toString();
```

Comment:
Instead of using toString use the EObject to compare see below for an example : 
private Property getSourceEndNaryAssociation(Association association, DDiagramElement view){
		final DEdge edge = (DEdge)view;
		final List<Property> members = association.getMemberEnds();

		final DNodeListSpec source = (DNodeListSpec)edge.getSourceNode();
		final List<EObject> sourceSemanticElements = source.getSemanticElements();
		for (final EObject sourceEObject : sourceSemanticElements) {
			for (final Property member : members) {
				if (sourceEObject.equals(member.getType())) {
					return member;
				}
			}
		}

		return null;
	}